homeassistant/components/xbox/media_player.py

```diff
@@ -0,0 +1,247 @@
+"""Xbox Media Player Support."""
+import logging
+import re
+from typing import List, Optional
+
+from xbox.webapi.api.client import XboxLiveClient
+from xbox.webapi.api.provider.catalog.const import HOME_APP_IDS, SYSTEM_PFN_ID_MAP
+from xbox.webapi.api.provider.catalog.models import AlternateIdType, Image, Product
+from xbox.webapi.api.provider.smartglass.models import (
+    PlaybackState,
+    PowerState,
+    SmartglassConsole,
+    SmartglassConsoleList,
+    SmartglassConsoleStatus,
+    VolumeDirection,
+)
+
+from homeassistant.components.media_player import MediaPlayerEntity
+from homeassistant.components.media_player.const import (
+    MEDIA_TYPE_APP,
+    MEDIA_TYPE_GAME,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_STEP,
+)
+from homeassistant.const import STATE_OFF, STATE_ON, STATE_PAUSED, STATE_PLAYING
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_XBOX = (
+    SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+    | SUPPORT_PREVIOUS_TRACK
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_PLAY
+    | SUPPORT_PAUSE
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_VOLUME_MUTE
+)
+
+XBOX_STATE_MAP = {
+    PlaybackState.Playing: STATE_PLAYING,
+    PlaybackState.Paused: STATE_PAUSED,
+    PowerState.On: STATE_ON,
+    PowerState.SystemUpdate: STATE_OFF,
+    PowerState.ConnectedStandby: STATE_OFF,
+    PowerState.Off: STATE_OFF,
+    PowerState.Unknown: None,
+}
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Xbox media_player from a config entry."""
+    client: XboxLiveClient = hass.data[DOMAIN][entry.entry_id]
+    consoles: SmartglassConsoleList = await client.smartglass.get_console_list()
+    async_add_entities(
```

Comment:
I was testing the PR now, and we should consider adding a debug log of the returned consoles here.